s2counter_core/src/main/scala/s2/counter/core/v2/ExactStorageGraph.scala

```diff
@@ -42,26 +42,58 @@ case class ExactStorageGraph(config: Config) extends ExactStorage {
   override def update(policy: Counter, counts: Seq[(ExactKeyTrait, ExactValueMap)]): Map[ExactKeyTrait, ExactValueMap] = {
     import scala.concurrent.ExecutionContext.Implicits.global
 
-    val (keyWithEq, reqJsLs) = toIncrementCountRequests(policy, counts).unzip(x => ((x._1, x._2), x._3))
-
-    val future = wsClient.url(s"$s2graphUrl/graphs/edges/incrementCount").post(Json.toJson(reqJsLs)).map { resp =>
-      resp.status match {
-        case HttpStatus.SC_OK =>
-          val respSeq = resp.json.as[Seq[RespGraph]]
+    val futureLs = {
+      for {
+        requests <- toIncrementCountRequests(policy, counts).grouped(10)
+        (keyWithEq, reqJsLs) = requests.unzip(x => ((x._1, x._2), x._3))
+      } yield {
+        wsClient.url(s"$s2graphUrl/graphs/edges/incrementCount").post(Json.toJson(reqJsLs)).map { resp =>
+          resp.status match {
+            case HttpStatus.SC_OK =>
+              val respSeq = resp.json.as[Seq[RespGraph]]
 
-          val keyWithEqResult = {
-            for {
-              ((key, eq), RespGraph(success, result)) <- keyWithEq.zip(respSeq)
-            } yield {
-              (key, (eq, result))
-            }
-          }.groupBy(_._1).mapValues{ seq => seq.map(_._2).toMap }
-          keyWithEqResult
-        case _ =>
-          throw new RuntimeException(s"update failed: $policy $counts")
+              for {
+                ((key, eq), RespGraph(success, result)) <- keyWithEq.zip(respSeq)
+              } yield {
+                (key, (eq, result))
+              }
+            case HttpStatus.SC_BAD_REQUEST =>
+              // logging
+              log.error(s"BAD_REQUEST: $policy $counts")
+              Nil
+            case rc: Int =>
+//              throw new RuntimeException(s"update failed($rc): $policy $counts")
```

Comment:
I want to keep going this spark job although an exception occurred. So, I had changed the code.